[WIP][GO-1] Prepare to Yaga 2.0.0

im-kulikov opened the pull request:
# Work around simplify

- comment documentation
- simplify auth-middleware
- fix migration
- Fix validation check errors
- remove unused imports
- cmd: create new project
- cmd: migrate create
- cmd: migrate up
- cmd: migrate down
- cmd: migrate version
- cmd: migrate cleanup
- fix: yaga/cli
- fix: yaga/migrate
- fix: examples/cli (debug)
- fix: CircleCI
- remove migrate/cmd
- bind params to struct

im-kulikov committed:
[GO-3] Work around simplify (#73)

* [GO-3] Work around simplify

* - comment documentation
- simplify auth-middleware
- fix migration

* Merge with last fix (#75)

* [GO-9] Fix validation check errors

* remove unused error

im-kulikov committed:
Merge branch 'master' into v2

pcherednichenko commented on middlewares/auth/model.go:
```diff
@@ -0,0 +1,75 @@
+		user  = &User{
+			Username:  username,
+			CreatedAt: time.Now(),
+			UpdatedAt: time.Now(),
```
Better  
`t := time.Now()`
And here to use `t`, so that the time is the same, in theory it may be different here

im-kulikov replied:
it's not new file.. https://github.com/cryptopay-dev/yaga/pull/76/files/7dabdf723032b223f7788a1c91b1f4daf9847cfa#diff-6b9fcb6effaad986aa964e5bd20f993aL30

pcherednichenko commented:
At all it's okay, but some "not new files" (in GitHub you just see new code and review it) and WIP status
___
@pcherednichenko At the moment, I need to understand if there are any comments, changes and / or suggestions for implementation

im-kulikov committed:
Merge branch 'master' into v2

im-kulikov committed:
Merge branch 'master' into v2

# Conflicts:
#	Gopkg.lock
#	cmd/yaga/project_example/app/app.go

im-kulikov committed:
small fix

im-kulikov committed:
set yaml version 2.1.1

im-kulikov committed:
update  Gopkg.toml

im-kulikov committed:
Merge branch 'master' of github.com:cryptopay-dev/yaga into v2

im-kulikov committed:
Merge branch 'master' of github.com:cryptopay-dev/yaga into v2

im-kulikov committed:
Merge branch 'master' of github.com:cryptopay-dev/yaga into v2

im-kulikov committed:
Merge branch 'master' into v2

jenchik committed:
[GO-85] binding default values (#91)

* [GO-85] binding default values

* [GO-85] added tests

im-kulikov committed:
Merge branch 'master' into v2

im-kulikov committed:
[GO-41] Refactor config (#97)

* [GO-41] Refactor config

- db / redis helpers
- config refactoring
- testdb from `config`
- fix / refactor code

* fix facepalm

* remove unused const

* remove empty line

im-kulikov committed:
Merge branch 'master' of github.com:cryptopay-dev/yaga into v2

jenchik committed:
[GO-76] added gracefull component (#96)

* [GO-76] added gracefull component

* [GO-76] changes

* [GO-76] change Notify
+ rename package

* [GO-76] fix example

* [GO-76] improvement package web

* [GO-76] fix previous merge

chapsuk committed:
Update processed signals for graceful shutdown

jenchik committed:
[GO-89] wrap errors (#99)

im-kulikov committed:
Merge branch 'master' of github.com:cryptopay-dev/yaga into v2

jenchik committed:
[GO-000] race access to redis-lock.Options.LockTimeout (#101)

im-kulikov committed:
[GO-000] Hide types (#103)

im-kulikov committed:
Merge branch 'master' into v2

im-kulikov committed:
Merge branch 'master' into v2

jenchik committed:
[GO-105] added Context for graceful (#106)

im-kulikov committed:
Merge branch 'master' into v2

im-kulikov committed:
[GO-000] Global logger (#109)

* [GO-000] Global logger

- global logger
- examples
- dep ensure -update

* - Gopkg.toml fix decimal

* - config init with ReadInConfig

* fix tests

* fix tests

* comments

im-kulikov committed:
[GO-000] Rewrite with global logger (#110)

* [GO-000] Rewrite with global logger

* cli.Run error when logger not inited

* log.Init...

* fix error

im-kulikov committed:
[GO-000] Global Logger (#114)

* [GO-000] Global Logger

- logger now not need config (os.Getenv -> LEVEL: prod, dev, nil, ...)
- remove log.Init()
- remove core
- add callerSkip
- fix order
- remove unused

im-kulikov committed:
[GO-000] Bump dependencies (#115)

im-kulikov committed:
start async, `bind` from config

im-kulikov committed:
fix examples

im-kulikov committed:
remove duplicate log message

im-kulikov committed:
[GO-000] Refactoring (#118)

* [GO-000] Refactoring

- speedup auth middleware tests
- refactoring model-package
- migrate `new` test case loop 10 to 2

* use `circleci/redis:alpine` image

* - refactoring `helpers/testdb`
- remove `godoenv`..

* - remove unused code

im-kulikov committed:
[GO-000] Locker interface with drivers (#117)

* [GO-000] Locker interface with redis driver

jenchik committed:
[GO-111] simplest workers v3 (#113)

chapsuk committed:
Metrics (#121)

* Added metrics package, common  metrics writer interface.
Implemented metrics provider for prometheus
* Added nop metrics provider

im-kulikov committed:
fix logger
